server/src/main/java/org/eclipse/lsp/cobol/service/delegates/completions/Keywords.java

```diff
@@ -17,14 +17,21 @@
 import com.google.inject.Singleton;
 
 import java.io.InputStream;
+import java.io.SequenceInputStream;
+import java.util.List;
 
 /** This class is a provider for Cobol keywords and their descriptions */
 @Singleton
 public class Keywords extends CompletionStorage {
   private static final String KEYWORDS_FILE_PATH = "LanguageKeywords.txt";
+  private static final String KEYWORDS_IDMS_FILE_PATH = "LanguageKeywordsIDMS.txt";
 
   @Override
-  protected InputStream getInputStream() {
-    return Keywords.class.getResourceAsStream(KEYWORDS_FILE_PATH);
+  protected InputStream getInputStream(List<String> dialectList) {
+    if (!dialectList.isEmpty() && dialectList.contains("IDMS")) {
```

Comment:
There is a constant for dialect name org.eclipse.lsp.cobol.core.engine.dialects.idms.IdmsDialect#NAME

Reply:
done